File: lib_common/src/main/java/com/wujia/lib_common/data/network/NetConfig.java
package com.wujia.lib_common.data.network;

import okhttp3.CookieJar;
import okhttp3.Interceptor;

/**
 * Created by xmren on 2018/5/4.
 */

public class NetConfig {
    public Interceptor[] mInterceptors;
    public CookieJar mCookieJar;
    public long connectTimeoutMills;
    public long readTimeoutMills;
    public boolean isHasLog;
    public boolean isUseRx = true;
    public String baseURL = "";
    public boolean isUseMultiBaseURL = true;
    public static boolean debug;
    public static String signKey;

    /**
     * add okhttp Interceptors
     * @param configInterceptors
     * @return
     */
    public NetConfig configInterceptors(Interceptor[] configInterceptors) {
        this.mInterceptors = configInterceptors;
        return this;
    }

    public NetConfig configisUseMultiBaseURL(boolean isUseMultiBaseURL) {
        this.isUseMultiBaseURL = isUseMultiBaseURL;
        return this;
    }

    public NetConfig configBaseURL(String baseURL) {
        this.baseURL = baseURL;
        return this;
    }

    /**
     * config cookieManager
     * @param mCookieJar
     * @return
     */
    public NetConfig configCookieJar(CookieJar mCookieJar) {
        this.mCookieJar = mCookieJar;
        return this;
    }


    public NetConfig configConnectTimeoutMills(long connectTimeoutMills) {
        this.connectTimeoutMills = connectTimeoutMills;
        return this;
    }

    public NetConfig configReadTimeoutMills(long readTimeoutMills) {
        this.readTimeoutMills = readTimeoutMills;
        return this;
    }

    public NetConfig configLogEnable(boolean isHasLog) {
        this.isHasLog = isHasLog;
        return this;
    }

    public NetConfig configIsUseRx(boolean isUseRx) {
        this.isUseRx = isUseRx;
        return this;
    }


}

File: lib_common/src/main/java/com/wujia/lib_common/data/network/NetConfig.kt
package com.wujia.lib_common.data.network

import okhttp3.CookieJar
import okhttp3.Interceptor

/**
 * Created by xmren on 2018/5/4.
 */

class NetConfig {
    var mInterceptors: Array<Interceptor>?=null
    var mCookieJar: CookieJar?=null
    var connectTimeoutMills: Long = 0
    var readTimeoutMills: Long = 0
    private var isHasLog: Boolean = false
    private var isUseRx = true
    var baseURL = ""
    private var isUseMultiBaseURL = true

    /**
     * add okhttp Interceptors
     *
     * @param configInterceptors
     * @return
     */
    fun configInterceptors(configInterceptors: Array<Interceptor>): NetConfig {
        this.mInterceptors = configInterceptors
        return this
    }

    fun configisUseMultiBaseURL(isUseMultiBaseURL: Boolean): NetConfig {
        this.isUseMultiBaseURL = isUseMultiBaseURL
        return this
    }

    /**
     * config cookieManager
     *
     * @param mCookieJar
     * @return
     */
    fun configCookieJar(mCookieJar: CookieJar): NetConfig {
        this.mCookieJar = mCookieJar
        return this
    }


    fun configConnectTimeoutMills(connectTimeoutMills: Long): NetConfig {
        this.connectTimeoutMills = connectTimeoutMills
        return this
    }

    fun configReadTimeoutMills(readTimeoutMills: Long): NetConfig {
        this.readTimeoutMills = readTimeoutMills
        return this
    }

    companion object {
        var debug: Boolean = false
        var signKey: String? = null
    }


}
